Integrations/Panorama/Panorama.py

```diff
@@ -55,6 +55,21 @@
 else:
     XPATH_OBJECTS = "/config/devices/entry/vsys/entry[@name=\'" + VSYS + "\']/"
 
+# setting template xpath relevant to panorama instances.
+if demisto.args() and demisto.args().get('template', None):
+    TEMPLATE = demisto.args().get('template')
+else:
+    TEMPLATE = demisto.params().get('template', None)
+if TEMPLATE:
+    if not DEVICE_GROUP or VSYS:
+        return_error('Template is only relevant for Panorama instances.')
+# setting network xpath relevant to FW or panorama management
+if DEVICE_GROUP:
+    XPATH_NETWORK = f'/config/devices/entry[@name=\'localhost.localdomain\']/template/entry[@name=\'{TEMPLATE}\']' \
+                    f'/config/devices/entry[@name=\'localhost.localdomain\']/network'
+else:
+    XPATH_NETWORK = "/config/devices/entry[@name='localhost.localdomain']/network"
+
```

Comment:
Accessing the arguments(apposed to parameters) in the beginning of the integration is confusing and anti-pattern. I think this code should be moved to a single function that gets the `template` argument(from args or params) and turns the XPATH_NETWORK. this function will be used by the relevant business logic functions that needs the XPATH_NETWORK value.